ansible/roles/kubernetes/tasks/debian.yml

```diff
@@ -97,6 +97,19 @@
 # sysctl net.bridge.bridge-nf-call-iptables=1 to pass bridged IPv4 traffic to iptables’ chains.
 # This is a requirement for some CNI plugins to work.
 # https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#letting-iptables-see-bridged-traffic
+- name: Load br_netfilter
+  modprobe:
+    name: br_netfilter
+    state: present
+
+- name: persist br_netfilter load through reboots
+  lineinfile:
```

Comment:
I'd prefer the copy module to be used and to write a file to `/etc/modules-load.d/br_netfilter.conf` as `/etc/modules` is very specific to Ubuntu only. The `/etc/modules-load.d` path is universal to all distributions using systemd (which also applies to Ubuntu). This would make it more versatile. Also, I'd prefer to see a `when: modprobe_br_netfilter.changed` statement as some distributions (like Arch) loads this module by default, making the file in `/etc/modules-load.d` redundant. The `when` statement would then make sure it only gets put there if the modprobe had to manually load the module.

Reply:
Adding to the reasoning here a little: Modifying `/etc/modules` is a bit invasive when we can simply just drop in configuration files to `/etc/modules-load.d`. This method is also more extensible in the case that we need to add more kernel modules in the future.

Adding in tons of `lineinfile` regex just seems a bit dirty to me. 

Another potentially more extensible solution would be to create a var listing the kernel modules needed, and using the `template` module to create the remote file. This would allow users to tweak a variable to add more modules.

Perhaps something like `/etc/modules-load.d/kubernetes_modules.conf` or the like.

Just a suggestion!

Reply:
I like that suggestion, but that does make it a bit harder to _only_ include modules in that file that are not already loaded by default. Although, I don't think it would be a big issue to explicitly load modules that are already implicitly loaded by default, but haven't tested this. As I am running on Arch, I can test it once the code changes have been made for this.

Also a note: if adding variables to a role (for example, a list variable for all the modules to load), then make sure to prefix it with the role name, e.g.: `kubernetes_kernel_modules`.

Reply:
I can't say for certain, but I can't imagine that there would be any detrimental effects from "double-loading" kernel modules. 

I see the `/etc/modules-load.d/kubernetes_modules.conf` as more of a centralized place to see what modules are enabled/needed by Kubernetes to run. Additionally, it makes it easier to track what modules are being loaded by our playbook vs any modules that the user may elect to add later.